configure.ac

```
@@ -443,11 +443,33 @@ if test "x$with_dl" != "xno"; then
   AC_CHECK_LIB(dl, dlopen)
 fi
 
+AC_DEFUN([LIBATOMIC_CHECKER],[
+  AC_MSG_CHECKING(whether linking $1 -latomic works)
+  AC_LINK_IFELSE([
+    AC_LANG_SOURCE([[
+      #include <atomic>
```

The whitespace prefix here will break/confuse some pre-compilers and make the test fail wrong.
It is annoying but the # does have to be the first character of the line for precompiler directives.

```suggestion
#include <atomic>
```

---

Done, however modern C/C++ standards [allow](https://stackoverflow.com/questions/27871083/does-character-have-to-be-at-the-start-of-a-line-in-the-c-preprocessor) whitespaces before preprocessor directives.  For example, this is stated in [gcc](https://gcc.gnu.org/onlinedocs/gcc-2.95.3/cpp_1.html#SEC3) and [microsoft](https://docs.microsoft.com/en-us/cpp/preprocessor/preprocessor-directives?view=vs-2019) preprocessor manuals.